### DISCUSSION TITLE
Is there any guide to create deb or rpm file for linux x64?

### DISCUSSION THREAD

**Discussion by mesaglam** [airsdk/Adobe-Runtime-Support#2136] [Guides & Tutorials]
Could you share a guide to create deb file or rpm file please?

**Comment by ajwfrost** [comment]
Currently the best option is to use the "-target bundle" option so that you end up with a folder containing the application, with the captive runtime. And then use something like `rpmbuild` to turn this into an rpm file for distribution.

The deb/rpm "native" targets in AIR aren't fully working - and actually all of the "native" installers rely on there being a suitable version of AIR already installed on the machine, if there's no version then I think they try to download and install it silently but that isn't working currently (you may still end up with AIR 32 though!) -> hence the recommended approach being to use the "bundle" target.

thanks

**Reply by mesaglam** [reply]
Thank you...

It would be great if it will as easy as the building dmg file for mac.

**Comment by myflashlab** [comment]
Packaging an executable .deb file for most linux systems is not very complicated but the information is very hard to be found. it took me a while to figure this out but the following should help you start: 

The main command is this
```
dpkg-deb --build --root-owner-group "Name of your software"
```

And before that, you need to setup a folder structure similar to the following. Your version might be different based on your needs but this is what I came up with and am happy for my needs.
```
Name of your software
-- DEBIAN
---- control (file)
---- postinst (file)
---- postrm (file)
-- usr
---- name of your software
------ include all the files from the captive build of your software with AIR SDK
---- share
------ appdata
-------- Name of your software.desktop.appdata.xml (file)
------ applications
-------- Name of your software.desktop
------ pixmaps
-------- Name of your software.png (file)
------ icons
-------- hicolor
---------- 16x16
------------ apps
-------------- Name of your software.png (file)
---------- 32x32
------------ apps
-------------- Name of your software.png (file)
---------- 36x36
------------ apps
-------------- Name of your software.png (file)
---------- 64x64
------------ apps
-------------- Name of your software.png (file)
---------- 72x72
------------ apps
-------------- Name of your software.png (file)
---------- 128x128
------------ apps
-------------- Name of your software.png (file)
---------- 192x192
------------ apps
-------------- Name of your software.png (file)
---------- 256x256
------------ apps
-------------- Name of your software.png (file)
---------- 512x512
------------ apps
-------------- Name of your software.png (file)
```

Finally, to know what content the above files should have, you may read here: https://www.debian.org/doc/debian-policy/

A quick tip is about these three files inside the `DEBIAN` folder
```
---- control (file) >> includes main setup details of the .dep package
---- postinst (file) >> an executable bash script to happen after installation
---- postrm (file) >> similar to above but a little bit different which I can't remember now :D 
```

cheers
